crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -1417,6 +1449,57 @@ impl EntityCommands<'_> {
     }
 }
 
+/// A wrapper around [`EntityCommands`] with convenience methods for working with a specified component type.
+pub struct EntityEntryCommands<'a, T> {
+    entity_commands: EntityCommands<'a>,
+    marker: PhantomData<T>,
+}
+
+impl<'a, T: Component> EntityEntryCommands<'a, T> {
+    /// Modify the component `T` using the the predicate `f`.
+    pub fn and_modify(mut self, f: impl FnOnce(Mut<T>) + Send + Sync + 'static) -> Self {
+        self.entity_commands = self
+            .entity_commands
+            .queue(move |mut entity: EntityWorldMut| {
+                if let Some(value) = entity.get_mut() {
+                    f(value);
+                }
+            });
+        self
+    }
+
+    /// Insert `default` into this entity, if `T` is not already present.
+    ///
+    /// See also [`EntityCommands::insert`], [`or_insert_with`](Self::or_insert_with), [`or_default`](Self::or_default)
+    #[track_caller]
+    pub fn or_insert(mut self, default: T) -> Self {
+        self.entity_commands = self
+            .entity_commands
+            .queue(insert(default, InsertMode::Keep));
+        self
+    }
+
+    /// Insert the value returned from `default` into this entity, if `T` is not already present.
+    ///
+    /// See also [`EntityCommands::insert`], [`or_insert`](Self::or_insert), [`or_default`](Self::or_default)
+    #[track_caller]
+    pub fn or_insert_with(self, default: impl Fn() -> T) -> Self {
+        self.or_insert(default())
+    }
+
+    /// Insert `T::default` into this entity, if `T` is not already present.
+    ///
+    /// See also [`EntityCommands::insert`], [`or_insert`](Self::or_insert), [`or_insert_with`](Self::or_insert_with)
+    #[track_caller]
+    pub fn or_default(self) -> Self
+    where
+        T: Default,
+    {
+        #[allow(clippy::unwrap_or_default)]
```

Comment:
This should be an `expect` with a reason now that that's stable.

Reply:
I changed it to expect but couldn't come up with a reason, do you think you could help me? or maybe it's fine without?

Reply:
Why doesn't clippy's suggestion work? Or rather, why is clippy firing here at all?